qcify.py

```diff
@@ -0,0 +1,128 @@
+
+    args = parser.parse_args()
+
+    qc_array = []
+    if not args.SID:
+        exit("You need to supply a sample-id")
+    # Load Picard-like output wgs-metrics / hs-metrics aln_metrics is_metrics dedup_metrics.txt gc_summary.txt 
+    if args.assay_metrics:
+        qc_array.append(read_picard_like_files(args.assay_metrics,"assay"))
+    if args.aln_metrics:
+        qc_array.append(read_picard_like_files(args.aln_metrics,"alignment"))
+    if args.is_metrics:
+         qc_array.append(read_picard_like_files(args.is_metrics,"insert"))
+    if args.dedup_metrics:
+         qc_array.append(read_picard_like_files(args.dedup_metrics,"dedup"))
+    if args.gc_summary:
+        qc_array.append(read_picard_like_files(args.gc_summary,"gc"))
+    # Load other data structures
+    #cov_metrics (--histogram_high 2000 or even higher for panel-data to allow for IQR / MEDIAN )
+    if args.cov_summary:
+        qc_array.append(coverage_calc(args.cov_summary,"coverage"))
+    if args.mq_metrics:
+        qc_array.append(read_vertical_picard(args.mq_metrics,"qualbycycle"))
+    if args.qd_metrics:
+        qc_array.append(read_vertical_picard(args.qd_metrics,"qualdist"))
+    with open(f"{args.SID}.json", 'w') as json_file:
+        json.dump(qc_array, json_file, indent=4)
+
+def read_picard_like_files(filename,category):
+    """
+    This function reads tabulated header/values outputs from picard
+    and creates key=value dicts instead
+    """
+    with open(filename, 'r') as file:
+        lines = file.readlines()
+        # Iterate through the lines to find the one starting with '#'
+        for i, line in enumerate(lines):
+            if line.startswith('#'):
+                # The next line after the one starting with '#' contains the header
+                headers_line = lines[i + 1].strip().split('\t')
+                break
+        data_line = lines[i + 2].strip().split('\t')
+    picard_dict = dict(zip(headers_line, data_line))
+    algo_dict = {
+        "software" : category,
+        "version" : "null",
+        "result": picard_dict
+    }
+    return algo_dict
+
+def coverage_calc(filename,category):
+    """
+    Accepts CoverageMetrics summary from sentieon.
+    Return coverage-thresholds at set values and coverage uniformity (IQR/MEDIAN) calcs
+    """
+    with open(filename, 'r') as file:
+        lines = file.readlines()
+        headers_line = lines[0].strip().split('\t')
+        sample_line = lines[1].strip().split('\t')
+    cov_dict = dict(zip(headers_line, sample_line))
+    # coverage uniformity
+    # interquartile range of coverage, if Q1 and Q3 are valid integers
+    if isinstance( cov_dict["granular_Q1"], int) and isinstance(cov_dict['granular_Q3'], int):
+        iqr = (int(cov_dict['granular_Q3']) - int(cov_dict['granular_Q1']))
+        cov_dict['coverage_uniformity'] = iqr / int(cov_dict['granular_median'])
+    else:
+        cov_dict['coverage_uniformity'] = "null"
```

Comment:
```suggestion
        cov_dict['coverage_uniformity'] = None
```

`None` is easier to test for down the line, compared to `"null"`, which is a "truthy" non-empty string.

Reply:
yeah I forgot the name for null in python, so I blanked :P